action/protocol/context.go

```diff
@@ -17,54 +17,64 @@ import (
 	"github.com/iotexproject/iotex-core/pkg/log"
 )
 
-// TipInfo contains the tip block information
-type TipInfo struct {
-	Height    uint64
-	Hash      hash.Hash256
-	Timestamp time.Time
-}
+type (
+	blockchainContextKey struct{}
 
-type blockchainContextKey struct{}
+	blockContextKey struct{}
 
-type blockContextKey struct{}
+	actionContextKey struct{}
 
-type actionContextKey struct{}
+	registryContextKey struct{}
 
-type registryContextKey struct{}
+	validationContextKey struct{}
```

Comment:
can this just stay in poll protocol now?

Reply:
fixed in latest push